src/lib/backoff.ts

```diff
@@ -0,0 +1,66 @@
+import { promisify } from 'util';
+
+const sleep = promisify(setTimeout);
+
+export type Options = {
+	attempt: number;
+	maxDelay: number;
+	minDelay: number;
+	maxRetries: number;
+	retryCallback?: (msg: string) => void;
```

Comment:
I would make the `retryCallback` signature as 
```
(timeToNextAttempt: number, e: Error, attempts: number) => void
``` 

So the caller can act on the error information (e.g. whether is a network or HTTP error) and update its internal state

Reply:
review new signature